src/scope.rs

```diff
@@ -160,6 +162,7 @@ impl Scope {
             },
             device: self.device.clone(),
             control_deps: self.control_deps.clone(),
+            kernel_label: "".to_string(),
```

Comment:
The `kernel_label` should be inherited from the parent scope (i.e. `kernel_label: self.kernel_label.clone()`) here and in the other methods below.

Reply:
Done.